core/src/test/java/org/apache/iceberg/deletes/TestRoaringPositionBitmap.java

```diff
@@ -0,0 +1,323 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.deletes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Set;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.relocated.com.google.common.io.Resources;
+import org.junit.jupiter.api.Test;
+
+public class TestRoaringPositionBitmap {
+
+  private static final long BITMAP_OFFSET = 0xFFFFFFFFL + 1L;
+  private static final long CONTAINER_OFFSET = Character.MAX_VALUE + 1L;
+  private static final Set<String> SUPPORTED_OFFICIAL_EXAMPLE_FILES =
+      ImmutableSet.of("64map32bitvals.bin", "64mapempty.bin", "64mapspreadvals.bin");
+
+  @Test
+  public void testAdd() {
+    RoaringPositionBitmap bitmap = new RoaringPositionBitmap();
+
+    bitmap.add(10L);
+    assertThat(bitmap.contains(10L)).isTrue();
+
+    bitmap.add(0L);
+    assertThat(bitmap.contains(0L)).isTrue();
+
+    bitmap.add(10L);
+    assertThat(bitmap.contains(10L)).isTrue();
+  }
+
+  @Test
+  public void testAddPositionsRequiringMultipleBitmaps() {
+    RoaringPositionBitmap bitmap = new RoaringPositionBitmap();
+
+    // construct positions that differ in their high 32-bit parts (i.e. keys)
+    long pos1 = ((long) 0 << 32) | 10L; // high = 0, low = 10
+    long pos2 = ((long) 1 << 32) | 20L; // high = 1, low = 20
+    long pos3 = ((long) 2 << 32) | 30L; // high = 2, low = 30
+    long pos4 = ((long) 100 << 32) | 40L; // high = 1000, low = 40
+
+    bitmap.add(pos1);
+    bitmap.add(pos2);
+    bitmap.add(pos3);
+    bitmap.add(pos4);
+
+    assertThat(bitmap.contains(pos1)).isTrue();
+    assertThat(bitmap.contains(pos2)).isTrue();
+    assertThat(bitmap.contains(pos3)).isTrue();
+    assertThat(bitmap.contains(pos4)).isTrue();
+
+    assertThat(bitmap.cardinality()).isEqualTo(4);
+
+    assertThat(bitmap.serializedSizeInBytes()).isGreaterThan(4);
+  }
+
+  @Test
+  public void testAddRange() {
```

Comment:
It might be good to test range across key boundary?

Reply:
Added a separate test below.